### PR TITLE
[MNT-21953] [MNT-22491] Clear rendition content data on content change. Prevent rendition from having contentHashCode without content

### PR DESCRIPTION
**Pull request by tiagosalvado10**
This PR has been raised regarding 2 issues.

**MNT-21953**

- It has been reported that some thumbnails couldn't be re-created after certain failures
- During the analysis it was noticed that there were nodes with, e.g., doclib renditions containing the rn:contentHashCode property but without the content itself
- In the scenario above, no matter how the rendition is requested again, it won't be re-created

This PR prevents a rendition from having the rn:contentHashCode property when there is no content. The rendition node is still created (without the hashCode and content) but new requests will be able to generate it.

**MNT-22491**

- It has been reported that uploading a new version (with transformation services down) will still display the previous preview
- Even starting the transformation services again won't show the new version preview but the old one instead

This PR approach consists in clearing the existing rendition content elements when a content change is detected.

### CODE REVIEW

**Review by alandavis — APPROVED** [review]
I like the changes that you have made as you have managed to do it without changing the original RenditionService2 approach. The risks are much lower as a result.

The changes to the older rendition services also make sense, fixing gaps in the attempt to support the Share UI handing off work to the newer RenditionService2 were possible.

**Comment by tiagosalvado10** [comment]
Added unit tests to PR:

- new method: RenditionService2Impl.clearRenditionContentData
- Rendition without content but with content hash code

File: repository/src/test/java/org/alfresco/repo/rendition2/RenditionService2IntegrationTest.java [hunk 1]
```diff
@@ -308,6 +309,99 @@ public void testUpgradeRenditionService() throws InterruptedException
...
+            transactionService.getRetryingTransactionHelper().doInTransaction(() ->
+            {
+                renditionService2.clearRenditionContentData(sourceNodeRef, DOC_LIB);
+                return null;
+            }), ADMIN);
+
+        // The rendition should have not content by now
+        assertNull("Rendition has content", nodeService.getProperty(renditionNodeRef, ContentModel.PROP_CONTENT));
+
+        // The rendition should have not a content hash code by now
```
Comment (alandavis) [inline repository/src/test/java/org/alfresco/repo/rendition2/RenditionService2IntegrationTest.java:348]: Change comment to: The rendition should not have a content hash code by now

File: repository/src/test/java/org/alfresco/repo/rendition2/RenditionService2IntegrationTest.java [hunk 2]
```diff
@@ -308,6 +309,99 @@ public void testUpgradeRenditionService() throws InterruptedException
...
+
+        // Call 'clearRenditionContentData' method directly to prove rendition content will be cleaned
+        AuthenticationUtil.runAs((AuthenticationUtil.RunAsWork<Void>) () ->
+            transactionService.getRetryingTransactionHelper().doInTransaction(() ->
+            {
+                renditionService2.clearRenditionContentData(sourceNodeRef, DOC_LIB);
+                return null;
+            }), ADMIN);
+
+        // The rendition should have not content by now
```
Comment (alandavis) [inline repository/src/test/java/org/alfresco/repo/rendition2/RenditionService2IntegrationTest.java:345]: Change comment to: The rendition should not have content by now

File: repository/src/test/java/org/alfresco/repo/rendition2/RenditionService2IntegrationTest.java [hunk 3]
```diff
@@ -25,6 +25,11 @@
  */
 package org.alfresco.repo.rendition2;
```
Comment (alandavis) [inline repository/src/test/java/org/alfresco/repo/rendition2/RenditionService2IntegrationTest.java:27]: Please change the Copyright year from 2018 to 2021

File: repository/src/test/java/org/alfresco/repo/rendition2/AbstractRenditionIntegrationTest.java [hunk 4]
```diff
@@ -25,7 +25,16 @@
  */
```
Comment (alandavis) [inline repository/src/test/java/org/alfresco/repo/rendition2/AbstractRenditionIntegrationTest.java:25]: Please change the Copyright year from 2018 to 2021

File: repository/src/main/java/org/alfresco/repo/rendition2/RenditionService2Impl.java [hunk 5]
```diff
@@ -538,9 +538,8 @@ else if (!nodeService.hasAspect(renditionNode, RenditionModel.ASPECT_RENDITION2)
                             }
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/repo/rendition2/RenditionService2Impl.java:538]: Please change the Copyright year from 2020 to 2021

File: repository/src/main/java/org/alfresco/repo/thumbnail/ThumbnailServiceImpl.java [hunk 6]
```diff
@@ -640,7 +640,7 @@ private boolean validateThumbnail(NodeRef thumbnailNode)
         boolean valid = true;
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/repo/thumbnail/ThumbnailServiceImpl.java:640]: Please change the Copyright year from 2017 to 2021

File: repository/src/main/java/org/alfresco/repo/rendition/RenditionServiceImpl.java [hunk 7]
```diff
@@ -683,6 +683,7 @@ else if (renditionDefinition2 != null)
                     log.debug("OnContentUpdate calling RenditionService2.render(\""+sourceNodeRef+"\", \""+renditionName+"\" so we switch to the new service.");
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/repo/rendition/RenditionServiceImpl.java:683]: Please change the Copyright year from 2016 to 2021

**Review by alandavis — APPROVED** [review]
Approved subject to a few comment changes.

If you change a Java file, please update the Copyright year in the header.

**Pull request merged**